app/src/main/kotlin/cz/covid19cz/erouska/ui/help/HelpVM.kt

```diff
@@ -13,4 +32,81 @@ class HelpVM @ViewModelInject constructor() : BaseVM() {
     fun openChatBot() {
         publish(HelpCommandEvent(HelpCommandEvent.Command.OPEN_CHATBOT))
     }
-}
\ No newline at end of file
+
+    fun searchQuery(query: String?, tvContent: String) {
+        searchJob?.cancel()
+
+        this.queryData.value = query?.trim() ?: ""
+
+        if (queryData.value.length >= 3) {
+            searchJob = viewModelScope.launch {
+                try {
+                    delay(400)
+
+                    val pattern = queryData.value
+                    val r = Pattern.compile(pattern, Pattern.CASE_INSENSITIVE)
+                    var result = AppConfig.helpMarkdown
+                    val m = r.matcher(result)
+                    val replaceList = arrayListOf<String>()
+
+                    while (m.find()) {
+                        replaceList.add(result.substring(m.start(0), m.end(0)))
+                    }
+                    searchResultCount.value = replaceList.size
+
+                    for (replaceString in replaceList.distinct()) {
+                        result = result.replace(replaceString, "[[${replaceString}]]")
+                    }
+
+                    content.value = result
+                    searchControlsEnabled.value = replaceList.isNotEmpty()
+
+                    lastMarkedIndex.value = tvContent.indexOf(queryData.value, ignoreCase = true)
+
+                } catch (cancelException: CancellationException) {
+                    L.w("job cancelled")
+                } catch (exception: Exception) {
```

Comment:
What other exception can occur? You should catch either more specific exception, or let the app crash.